lms/djangoapps/grades/api/tests/test_views.py

```diff
@@ -0,0 +1,259 @@
+        assert resp.status_code == status.HTTP_404_NOT_FOUND
+        self.assertIn('error_code', resp.data)  # pylint: disable=no-member
+        self.assertEqual(
+            resp.data['error_code'],  # pylint: disable=no-member
+            'invalid_course_key'
+        )
+
+    def test_couse_does_not_exist(self):
+        """
+        Test if the course key is valid but the course does not exist
+        """
+        # from nose.tools import set_trace; set_trace()
+        base_url = reverse(
+            self.namespaced_url,
+            kwargs={
+                'course_id': 'course-v1:MITx+8.MechCX+2014_T1',
+            }
+        )
+        url = "{0}?username={1}".format(base_url, self.student.username)
+        resp = self.client.get(url)
+        assert resp.status_code == status.HTTP_404_NOT_FOUND
+        self.assertIn('error_code', resp.data)  # pylint: disable=no-member
+        self.assertEqual(
+            resp.data['error_code'],  # pylint: disable=no-member
+            'user_or_course_does_not_exist'
+        )
+
+    def test_no_grade(self):
+        """
+        Test the grade for a student who has not answered any test
+        """
+        resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        expected_data = [{
+            'username': self.student.username,
+            'letter_grade': None,
+            'percent': 0.0,
+            'course_key': str(self.course_key),
+            'passed': False
+        }]
+        self.assertEqual(resp.data, expected_data)  # pylint: disable=no-member
+
+    def test_grade(self):
+        """
+        Test that the user gets her grade
+        """
+        with patch('lms.djangoapps.grades.new.subsection_grade.get_score', return_value=(2, 5)):
+            resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        expected_data = [{
+            'username': self.student.username,
+            'letter_grade': None,
+            'percent': 0.4,
+            'course_key': str(self.course_key),
+            'passed': False
+        }]
+        self.assertEqual(resp.data, expected_data)  # pylint: disable=no-member
+
+    def test_perfect_grade(self):
+        """
+        Test that the user gets the perfect grade if she gets the max score in each problem
+        """
+        with patch('lms.djangoapps.grades.new.subsection_grade.get_score', return_value=(5, 5)):
+            resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        expected_data = [{
+            'username': self.student.username,
+            'letter_grade': 'Pass',
+            'percent': 1.0,
+            'course_key': str(self.course_key),
+            'passed': True
+        }]
+        self.assertEqual(resp.data, expected_data)  # pylint: disable=no-member
```

Comment:
Nit: If you're up for it, `test_grade` and `test_perfect_grade` contain very similar logic and only differ in the value of a few parameters. They would be good candidates for `ddt.unpack`. See https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/api_admin/tests/test_forms.py#L21 as an example.